comms/dht/src/discovery/requester.rs

```diff
@@ -103,21 +64,25 @@ impl DhtDiscoveryRequester {
         }
     }
 
-    pub async fn discover_peer(
-        &mut self,
-        dest_public_key: Box<CommsPublicKey>,
-        dest_node_id: Option<NodeId>,
-        destination: NodeDestination,
-    ) -> Result<Peer, DhtDiscoveryError>
-    {
+    /// Initiate a peer discovery
+    ///
+    /// ## Arguments
+    /// - `dest_public_key` - The public key of he recipient used to create a shared ECDH key which in turn is used to
+    /// encrypt the discovery message
+    /// - `destination` - The `NodeDestination` to use in the DhtHeader when sending a discovery message.
+    ///    - `Unknown` destination will maintain complete privacy, the trade off is that discovery needs to propagate
+    ///      the entire network to reach the destination and so may take longer
+    ///    - `NodeId` Instruct propagation nodes to direct the message to peers closer to the given NodeId. The `NodeId`
+    ///      may be directed to a region close to the real destination (somewhat private) or directed at a particular
+    ///      node (not private)
+    ///    - `PublicKey` if any node on the network knows this public key, the message will be directed to that node.
+    ///      This sacrifices privacy for more efficient discovery in terms of network bandwidth and may result in
+    ///      quicker discovery times.
```

Comment:
These doc strings need to be updated.

Reply:
Perhaps I should add the node destination back in - it's a quick change

Reply:
...In a subsequent PR :)